lib/json_info_codec.dart

```diff
@@ -20,22 +20,28 @@ class JsonToAllInfoConverter extends Converter<Map<String, dynamic>, AllInfo> {
 
     var result = new AllInfo();
     var elements = json['elements'];
-    result.libraries
-        .addAll((elements['library'] as Map).values.map(parseLibrary));
-    result.classes.addAll((elements['class'] as Map).values.map(parseClass));
-    result.functions
-        .addAll((elements['function'] as Map).values.map(parseFunction));
+    // TODO(srawlins): Since only the Map values are being extracted below,
+    // replace `as` with `cast` when `cast` becomes available in Dart 2.0:
+    //
+    //     .addAll(elements['library'].values.cast<Map>().map(parseLibrary));
+    result.libraries.addAll(
+        (elements['library'] as Map<dynamic, Map>).values.map(parseLibrary));
```

Comment:
I believe these cast will fail at runtime.

These maps are created from json parsing, which I believe creates maps as `Map<dynamic, dynamic>` and lists as `List<dynamic>`

I'm guessing that changing this to `Map<dynamic, dynamic>` is equivalent to what we have today, correct? If so, I believe we have a few options:

* option A: wrap the closure to avoid the fuzzy arrow (this has an implicit cast):
```dart
(elements['library'] as Map).values.map((l) => parseLibrary(l));
```

* option B: same, but with the cast more explicit
```dart
(elements['library'] as Map).values.map((l) => parseLibrary(l as Map));
```

* option C: make the functions take dynamic and cast the parameter at the callee:
```dart
parseLibrary(dynamic jsonArg) {
  Map json = jsonArg;
  ...
}
```

I prefer option A.